crates/kernel/src/tasks/scheduler.rs

```diff
@@ -678,10 +697,21 @@ impl Scheduler {
 
                 let input_request_id = Uuid::new_v4();
                 {
-                    let Some(mut task) = self.tasks.get_mut(&task_id) else {
+                    let mut tasks = self.tasks.lock().unwrap();
+                    let Some(task) = tasks.get_mut(&task_id) else {
                         warn!(task_id, "Task not found for input request");
                         return None;
                     };
+                    // Commit the session (not DB transaction) to make sure current output is
+                    // flushed up to the prompt point.
+                    let Ok(()) = task.session.commit() else {
```

Comment:
I *think* this is unrelated?

Reply:
Actually intentional, just didn't mention in commit